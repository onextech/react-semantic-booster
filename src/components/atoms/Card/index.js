import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card as suiCard } from 'semantic-ui-react';
import { getCustomClassName, subtractObject } from '../../../utils/helpers';


const basicClassName = 'basic';
const stretchedClassName = 'stretched';

const StyledCard = styled(suiCard)`
  &.ui.card {
    &.${basicClassName} { box-shadow: none; }
    &.${stretchedClassName} { height: 100%; }
  }
`;

const Card = (card) => {
  // 1. Define custom props for this component
  const customCardProps = {
    basic: basicClassName,
    stretched: stretchedClassName,
  };
  // 1.5. Define default props
  const defaultCardProps = {
    basic: false,
    stretched: false,
  };
  const allCardProps = { ...defaultCardProps, ...card };
  // 2. Render the custom class names
  const cardClassName = card &&
    getCustomClassName(customCardProps, allCardProps);
  // 3. Clean up custom props from main prop set to avoid prop clashing
  const semanticCardProps = card &&
    subtractObject({ ...defaultCardProps, ...customCardProps }, allCardProps);

  return (
    <StyledCard {...semanticCardProps} className={cardClassName} />
  );
};

Card.Content = suiCard.Content;
Card.Description = suiCard.Description;

Card.propTypes = {
  card: PropTypes.object, // semantic-ui card props
  basic: PropTypes.bool,
  stretched: PropTypes.bool,
};

export default Card;
