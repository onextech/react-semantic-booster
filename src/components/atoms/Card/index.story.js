import React from 'react';
import { storiesOf } from '@storybook/react';
import { Grid } from 'semantic-ui-react';
import Card from '.';

storiesOf('Card', module)
  .add('Basic', () => (
    <Grid container>
      <Grid.Row>
        <Grid.Column>
          <Card basic>
            <br /><br /><br /><br /><br />
            Hello World.
            <br /><br /><br /><br /><br />
          </Card>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  ))
  .add('Stretched', () => (
    <Grid container>
      <Grid.Row>
        <Grid.Column>
          <Card stretched>
            Hello World.
          </Card>
        </Grid.Column>
        <Grid.Column>
          <Card stretched>
            <br /><br /><br /><br /><br />
            Hello World.
            <br /><br /><br /><br /><br />
          </Card>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  ));
