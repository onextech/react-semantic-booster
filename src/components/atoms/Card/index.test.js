import React from 'react';
import renderer from 'react-test-renderer';
import Card from '.';


describe('<Card>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Card />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
