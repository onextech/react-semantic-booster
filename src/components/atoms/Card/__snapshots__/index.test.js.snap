// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card> renders correctly 1`] = `
<div
  className="ui card sc-bdVaJa jrjGOD"
  href={undefined}
  onClick={[Function]}
/>
`;
